tests/bgp/test_bgp_fact.py

```diff
@@ -9,7 +9,13 @@ def test_bgp_facts(duthosts, enum_dut_hostname, enum_asic_index):
     """compare the bgp facts between observed states and target state"""
 
     duthost = duthosts[enum_dut_hostname]
-    bgp_facts =duthost.bgp_facts(instance_id=enum_asic_index)['ansible_facts']
+
+    # Check if duthost is 'supervisor' card based on 'type' defined in inventory, and skip the test if dealing with supervisor card.
+    host_vars = duthost.sonichost.host.options["inventory_manager"].get_host(duthost.hostname).get_vars()
+    if 'type' in host_vars and host_vars['type'] == 'supervisor':
+        pytest.skip("bgp_facts not valid on supervisor card '%s'" % enum_dut_hostname)
```

Comment:
@sanmalho-git  It looks like we may frequently need to check the type of duthost. The methods like `DutHosts._is_supervisor_node` and `DutHosts._is_frontend_node` should be public. And definitions of these methods should be moved to DutHost class. Do you agree? Can you do the refactor in this PR?

Reply:
good suggestion - committed the changes